File: app/src/main/java/com/fractalwrench/crazycats/injection/ActivityScope.java
package com.fractalwrench.crazycats.injection;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Scope;

/**
 * This Scope identifies that the dependencies should only be retained for the lifecycle of an
 * activity
 */
@Scope
@Retention(RetentionPolicy.RUNTIME)
public @interface ActivityScope {}

File: app/src/main/java/com/fractalwrench/crazycats/injection/ActivityScope.kt
package com.fractalwrench.crazycats.injection

import javax.inject.Scope

/**
 * This Scope identifies that the dependencies should only be retained for the lifecycle of an
 * activity
 */
@Scope
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class ActivityScope
